staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/width_test.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+)
+
+func TestWorkEstimator(t *testing.T) {
+	tests := []struct {
+		name          string
+		requestURI    string
+		requestInfo   *apirequest.RequestInfo
+		counts        map[string]int64
+		countErr      error
+		seatsExpected uint
+	}{
+		{
+			name:          "request has no RequestInfo",
+			requestURI:    "http://server/apis/v1/foos/",
+			requestInfo:   nil,
+			seatsExpected: maximumSeats,
+		},
+		{
+			name:       "request verb is not list",
+			requestURI: "http://server/apis/v1/foos/",
+			requestInfo: &apirequest.RequestInfo{
+				Verb: "get",
+			},
+			seatsExpected: minimumSeats,
+		},
+		{
+			name:       "request verb is list, conversion to ListOptions returns error",
+			requestURI: "http://server/apis/v1/foos/1?limit=invalid",
```

Comment:
the path here is inconsistent with the requestInfo, right?

Reply:
yes, i corrected the paths according to kube api convention.